servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpLoadBalancerFactory.java

```diff
@@ -102,13 +157,48 @@ private Builder(
             this.strategy = strategy;
         }
 
+        /**
+         * Sets a {@link Function} that is invoked for each received response to check whether the response should be
+         * considered as an error, and if so, of what {@link ErrorClass class}.
+         * <p>
+         * <strong>Note: </strong>Enabling this feature has no effect to {@link LoadBalancer}s that are not evaluating
+         * it.
+         *
+         * @param peerResponseErrorClassifier A {@link Function} to check if a received response should be considered an
+         * error. Returning {@code null} from that {@link Function}, means the response is considered as successful.
+         * @return {@code this}.
+         */
+        public Builder<ResolvedAddress> peerResponseErrorClassifier(
+                final Function<HttpResponseMetaData, ErrorClass> peerResponseErrorClassifier) {
+            this.peerResponseErrorClassifier = peerResponseErrorClassifier;
+            return this;
+        }
+
+        /**
+         * Sets a {@link Function} that is invoked for each {@link Throwable} occurred in the flow, to classify it to an
+         * {@link ErrorClass}.
+         * <p>
+         * <strong>Note: </strong>Enabling this feature has no effect to {@link LoadBalancer}s that are not evaluating
+         * it.
+         *
+         * @param errorClassifier A {@link Function} to check if a received response should be considered an
+         * error.
+         * @return {@code this}.
+         */
+        public Builder<ResolvedAddress> errorClassifier(
```

Comment:
Any chance we can defer adding these methods until we have more confidence in new API? If we add them now, it will be harder to remove/change them later.

I'm ok to consider a dirty hack of making them pkg private. If you will ever need it in any use-case, you can create an external utility for `io.servicetalk.http.netty` package to get access to these methods.

Reply:
Agreed, removed the public APIs and left the defaults for the wiring. We can add the APIs back when we are done with integration evaluation.